### ISSUE TITLE
DefaultConnectionContext.builder().skipSslValidation(true) no longer appears to work?

### ISSUE THREAD

**Issue by mchitwood** [cloudfoundry/cf-java-client#602]
I upgraded from 2.0.3 to 2.1.0 and receive a null exception when I try to create a `ReactorCloudFoundryClient`.

**Comment by nebhale** [comment]
Can you please include the stack of the error?  Our testing disables SSL validation in many of our environments, so we believe that it works as it always has.

**Comment by mchitwood** [comment]
java.nio.channels.UnresolvedAddressException
	at sun.nio.ch.Net.checkAddress(Unknown Source)
	at sun.nio.ch.SocketChannelImpl.connect(Unknown Source)
	at io.netty.channel.socket.nio.NioSocketChannel.doConnect(NioSocketChannel.java:331)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.connect(AbstractNioChannel.java:254)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.connect(DefaultChannelPipeline.java:1266)
	at io.netty.channel.AbstractChannelHandlerContext.invokeConnect(AbstractChannelHandlerContext.java:556)
	at io.netty.channel.AbstractChannelHandlerContext.connect(AbstractChannelHandlerContext.java:541)
	at io.netty.channel.AbstractChannelHandlerContext.connect(AbstractChannelHandlerContext.java:523)
	at io.netty.channel.DefaultChannelPipeline.connect(DefaultChannelPipeline.java:985)
	at io.netty.channel.AbstractChannel.connect(AbstractChannel.java:255)
	at io.netty.bootstrap.Bootstrap$3.run(Bootstrap.java:252)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:418)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:454)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at java.lang.Thread.run(Unknown Source)

**Comment by nebhale** [comment]
I've just tried to connect to the address listed there and cannot connect to it:

```
wlan-10-81-130-97 master: telnet api.system.asv-pr.ice.predix.io 443
Trying 65.216.161.18...

...
```

I'd believe that this is behaving as expected and that endpoint isn't actually listening.

**Comment by mchitwood** [comment]
This service is not accessible from the Internet.

**Comment by nebhale** [comment]
Can you please post over the output of https://api.system.asv-pr.ice.predix.io/v2/info?

**Comment by mchitwood** [comment]
Here is the version information from the above output.  Let me know if there is a specific value you are looking for.  Thanks!  

"min_cli_version":null,"min_recommended_cli_version":null,"api_version":"2.62.0"

**Comment by mchitwood** [comment]
I appear to be get a connection object when I comment out the ssl validation directive but I receive the same stack trace.  I have rolled back to 2.0.4 and the code works without any issues.

        DefaultConnectionContext connectionContext = DefaultConnectionContext.builder()
            .apiHost(uri.getHost())
            .port(443)
            .keepAlive(false)
//          .skipSslValidation(true)
            .proxyConfiguration(getProxyConfiguration())
            .build();

**Comment by nebhale** [comment]
RIght, I believe that the issue is around how DNS is resolved when using a proxy.  It's going to take a bit of time for us to build an environment that can replicate this problem, but we'll take a look.

**Comment by nebhale** [comment]
@adrian-herscu The new re-occurance of #479, is being tracked here.

**Comment by adrian-herscu** [comment]
So... Is there any known working version?

**Comment by nebhale** [comment]
Anything in the 2.0.x line.

**Comment by adrian-herscu** [comment]
`origin/2.0.x` fails to build:

```
[ERROR] Failed to execute goal com.squareup.wire:wire-maven-plugin:2.2.0:generat
e-sources (default) on project cloudfoundry-client: Wire Plugin: Failure compili
ng proto sources. Failed to locate envelope.proto in [C:\Users\212600552\project
s\cf-java-client\cloudfoundry-client\..\vendor\dropsonde-protocol\events] -> [He
lp 1]
```

**Comment by nebhale** [comment]
@adrian-herscu You don't need to build the 2.0.x line, there are already a [number of releases](http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22org.cloudfoundry%22%20AND%20a%3A%22cloudfoundry-java-client%22) of it.  If you do want to build it, you need to follow [the instructions](https://github.com/cloudfoundry/cf-java-client#development) laid out in the README.

**Comment by adrian-herscu** [comment]
Silly me... Didn't see those before...
Anyway, now trying with 2.0.4.RELEASE and getting:

```
java.lang.IndexOutOfBoundsException: Source emitted more than one item
	at reactor.core.publisher.MonoSingle$SingleSubscriber.onNext(MonoSingle.java:124)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drainLoop(FluxFlatMap.java:661)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drain(FluxFlatMap.java:541)
	at reactor.core.publisher.FluxFlatMap$FlatMapInner.onSubscribe(FluxFlatMap.java:1072)
	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:95)
	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:59)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:385)
	at reactor.core.publisher.MonoFlatMap$FlattenSubscriber$InnerSubscriber.onNext(MonoFlatMap.java:194)
	at reactor.core.publisher.FluxFlattenIterable$FlattenIterableSubscriber.drainAsync(FluxFlattenIterable.java:358)
	at reactor.core.publisher.FluxFlattenIterable$FlattenIterableSubscriber.drain(FluxFlattenIterable.java:587)
	at reactor.core.publisher.FluxFlattenIterable$FlattenIterableSubscriber.onNext(FluxFlattenIterable.java:216)
	at reactor.core.publisher.FluxBuffer$BufferExactSubscriber.onComplete(FluxBuffer.java:179)
	at reactor.core.publisher.FluxConcatArray$ConcatArraySubscriber.onComplete(FluxConcatArray.java:177)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.checkTerminated(FluxFlatMap.java:801)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.innerComplete(FluxFlatMap.java:921)
	at reactor.core.publisher.FluxFlatMap$FlatMapInner.onComplete(FluxFlatMap.java:1107)
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onComplete(Operators.java:1198)
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onComplete(FluxPeekFuseable.java:217)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onComplete(FluxMapFuseable.java:152)
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:894)
	at reactor.core.publisher.MonoThenApply$MonoThenApplyManager$SecondSubscriber.onNext(MonoThenApply.java:201)
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:122)
	at reactor.core.publisher.FluxUsing$UsingFuseableSubscriber.onNext(FluxUsing.java:314)
	at reactor.core.publisher.FluxFilterFuseable$FilterFuseableSubscriber.onNext(FluxFilterFuseable.java:108)
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableConditionalSubscriber.onNext(FluxPeekFuseable.java:368)
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:892)
	at reactor.core.publisher.MonoReduce$ReduceSubscriber.onComplete(MonoReduce.java:149)
	at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:144)
	at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.onComplete(FluxSubscribeOn.java:160)
	at reactor.ipc.netty.common.NettyChannelHandler$InboundSink.complete(NettyChannelHandler.java:654)
	at reactor.ipc.netty.http.NettyHttpClientHandler.postRead(NettyHttpClientHandler.java:184)
	at reactor.ipc.netty.http.NettyHttpClientHandler.channelRead(NettyHttpClientHandler.java:135)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:350)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:435)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:267)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:250)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:350)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:350)
	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:233)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:350)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1066)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:900)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:411)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:248)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:350)
	at io.netty.handler.proxy.ProxyHandler.channelRead(ProxyHandler.java:253)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:350)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:435)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:252)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:350)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:372)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:358)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:129)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:610)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:551)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:465)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:437)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by adrian-herscu** [comment]
By the end of the day I tried all 2.0.x.RELEASE's and got same stack trace.
The 2.0.0.RELEASE has a different API that 2.0.1.RELEASE and my code didn't compile with it.
Any suggestion?

**Comment by nebhale** [comment]
@adrian-herscu It's not clear what you were trying to do that resulted in this error, but it's unlikely that it has anything to do with the subject of this issue.  Please open another issue with a proper bug report in it, if you think there is a problem.

**Comment by adrian-herscu** [comment]
My fault, I initialized it with wrong organization and space.

**Comment by nebhale** [comment]
We've isolated this issue and it's in the `reactor-netty` project.  It will require a change to that library which won't make the `2.4.0.RELEASE`, but as soon as it's available we'll do another release with it.

**Comment by nebhale** [comment]
The latest snapshot `2.5.0.BUILD-SNAPSHOT` should fix this issue.  If you can find the time, please take a look at it to confirm that we've solved your problem.  Note that `reactor-core` should be `3.0.5.RELEASE` and `reactor-netty` should be `0.6.1.RELEASE` with this snapshot.

**Comment by mchitwood** [comment]
Thanks Ben.

We will try to test this today.

**Comment by mchitwood** [comment]
This still does not appear to be working for us.

2017-02-23 15:44:38.467  WARN 11036 --- [enerContainer-3] cloudfoundry-client.trust                : Trusting SSL Certificate for api.system.asv-pr.ice.predix.io:443
reactor.core.Exceptions$ReactiveException: java.net.UnknownHostException: api.system.asv-pr.ice.predix.io
                at reactor.core.Exceptions.propagate(Exceptions.java:219)
                at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:126)
                at reactor.core.publisher.Mono.blockMillis(Mono.java:1231)
                at reactor.core.publisher.Mono.block(Mono.java:1209)
                at org.cloudfoundry.reactor.util.DefaultSslCertificateTruster.getUntrustedCertificates(DefaultSslCertificateTruster.java:150)
                at org.cloudfoundry.reactor.util.DefaultSslCertificateTruster.trust(DefaultSslCertificateTruster.java:84)
                at org.cloudfoundry.reactor._DefaultConnectionContext.lambda$trust$13(_DefaultConnectionContext.java:229)
                at java.util.Optional.ifPresent(Optional.java:159)
                at org.cloudfoundry.reactor._DefaultConnectionContext.trust(_DefaultConnectionContext.java:229)
                at org.cloudfoundry.reactor._DefaultConnectionContext.getRoot(_DefaultConnectionContext.java:134)
                at org.cloudfoundry.reactor.DefaultConnectionContext.access$1901(DefaultConnectionContext.java:24)
                at org.cloudfoundry.reactor.DefaultConnectionContext$InitShim.getRoot(DefaultConnectionContext.java:160)
                at org.cloudfoundry.reactor.DefaultConnectionContext.<init>(DefaultConnectionContext.java:72)
                at org.cloudfoundry.reactor.DefaultConnectionContext.<init>(DefaultConnectionContext.java:24)
                at org.cloudfoundry.reactor.DefaultConnectionContext$Builder.build(DefaultConnectionContext.java:887)
                at com.ge.aviation.paasport.provisioning.CFClient.getConnectionContext(CFClient.java:172)
                at com.ge.aviation.paasport.provisioning.CFClient.getCloudFoundryOperations(CFClient.java:135)
                at com.ge.aviation.paasport.provisioning.CFAppAgent.create0(CFAppAgent.java:171)
                at com.ge.aviation.paasport.provisioning.CFAppAgent.create(CFAppAgent.java:154)
                at com.ge.aviation.paasport.provisioning.AgentDispatcher.create(AgentDispatcher.java:439)
                at com.ge.aviation.paasport.provisioning.AgentDispatcher.dispatch(AgentDispatcher.java:286)
                at com.ge.aviation.paasport.provisioning.AgentActionListener.onMessage(AgentActionListener.java:64)
                at org.springframework.jms.listener.AbstractMessageListenerContainer.doInvokeListener(AbstractMessageListenerContainer.java:746)
                at org.springframework.jms.listener.AbstractMessageListenerContainer.invokeListener(AbstractMessageListenerContainer.java:684)
                at org.springframework.jms.listener.AbstractMessageListenerContainer.doExecuteListener(AbstractMessageListenerContainer.java:651)
                at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.doReceiveAndExecute(AbstractPollingMessageListenerContainer.java:315)
                at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.receiveAndExecute(AbstractPollingMessageListenerContainer.java:253)
                at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.invokeListener(DefaultMessageListenerContainer.java:1164)
                at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.executeOngoingLoop(DefaultMessageListenerContainer.java:1156)
                at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.run(DefaultMessageListenerContainer.java:1053)
                at java.lang.Thread.run(Thread.java:745)
                Suppressed: java.lang.Exception: #block terminated with an error
                                at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:127)
                                ... 29 more
Caused by: java.net.UnknownHostException: api.system.asv-pr.ice.predix.io
                at java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
                at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:928)
                at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1323)
                at java.net.InetAddress.getAllByName0(InetAddress.java:1276)
                at java.net.InetAddress.getAllByName(InetAddress.java:1192)
                at java.net.InetAddress.getAllByName(InetAddress.java:1126)
                at java.net.InetAddress.getByName(InetAddress.java:1076)
                at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:146)
                at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:143)
                at java.security.AccessController.doPrivileged(Native Method)
                at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:143)
                at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
                at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:62)
                at io.netty.resolver.InetNameResolver.resolve(InetNameResolver.java:59)
                at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
                at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
                at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
                at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
                at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:208)
                at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:49)
                at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:188)
                at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:174)
                at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:507)
                at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:481)
                at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:420)
                at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
                at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:82)
                at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:897)
                at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:508)
                at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:419)
                at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:478)
                at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
                at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
                at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:445)
                at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)

**Comment by nebhale** [comment]
And which versions of `cloudfoundry-java-client-reactor` (explicit snapshot number if you can determine it) and `reactor-netty` are you using?

**Comment by mchitwood** [comment]
Here are the entries in my pom.xml.

        <dependency>
            <groupId>org.cloudfoundry</groupId>
            <artifactId>cloudfoundry-client-reactor</artifactId>
            <version>2.5.0.BUILD-SNAPSHOT</version>
        </dependency>
        <dependency>
            <groupId>org.cloudfoundry</groupId>
            <artifactId>cloudfoundry-operations</artifactId>
            <version>2.5.0.BUILD-SNAPSHOT</version>
        </dependency>
        <dependency>
            <groupId>io.projectreactor</groupId>
            <artifactId>reactor-core</artifactId>
            <version>3.0.5.RELEASE</version>
        </dependency>
        <dependency>
            <groupId>io.projectreactor.ipc</groupId>
            <artifactId>reactor-netty</artifactId>
            <version>0.6.1.RELEASE</version>
        </dependency>

**Comment by nebhale** [comment]
@mchitwood The [fix has been made to `reactor-netty`](https://github.com/reactor/reactor-netty/commit/bd7bcf837fa4bd5df7ba9c2f87b5a200c4145777) and we'll be waiting on a `0.6.2.RELEASE` shortly.  If you're feeling really adventurous, you can (as of today) run our `2.5.0.BUILD-SNAPSHOT` with `reactor-netty 0.6.2.BUILD-SNAPSHOT`.  We won't release that way (it'll be `reactor-netty 0.6.1.RELEASE`) but depending on your level of comfort, it's available to you now.

**Comment by mchitwood** [comment]
Thanks Ben!
